common/achievement/util.ts

```diff
@@ -94,14 +94,20 @@ export async function getBucketContext(userID: string, relation?: string): Promi
 }
 
 export function transformPrismaJson(user: User, json: Prisma.JsonValue): AchievementContextType | null {
-    if (!json['match'] && !json['subcourse']) {
+    const keys = Object.keys(json);
+    if (!keys) {
         return null;
     }
     const transformedJson: AchievementContextType = {
         user: user,
         match: json['match'] ? json['match'] : undefined,
         subcourse: json['subcourse'] ? json['subcourse'] : undefined,
     };
+    keys.forEach((key) => {
+        if (key !== 'match' && key !== 'subcourse') {
```

Comment:
Can't we just copy all the keys and remove line 103 + 104? They are basically doing the same thing.

Reply:
this would then require us to change the AchievementContextType. I set match and subcourse to optional values.